server/src/main/java/org/elasticsearch/action/datastreams/GetDataStreamAction.java

```diff
@@ -269,26 +270,6 @@ public DataStreamInfo(
                 this.maximumTimestamp = maximumTimestamp;
             }
 
-            @SuppressWarnings("unchecked")
-            DataStreamInfo(StreamInput in) throws IOException {
-                this.dataStream = DataStream.read(in);
-                this.failureStoreEffectivelyEnabled = in.getTransportVersion()
-                    .onOrAfter(TransportVersions.FAILURE_STORE_ENABLED_BY_CLUSTER_SETTING)
-                        ? in.readBoolean()
-                        : dataStream.isFailureStoreExplicitlyEnabled(); // Revert to the behaviour before this field was added
```

Comment:
I think this behavior might be getting lost with the serialization going away. Should this be added back somewhere?

Reply:
I'm not sure I'm following. Are you referring to the fallback behavior of `dataStream.isFailureStoreExplicitlyEnabled()`? Why would we care about that? Any node that has this version is going to use:
https://github.com/elastic/elasticsearch/blob/eadffcb4b2cfc02395402a3cb35e44523773fd7d/modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/TransportGetDataStreamsAction.java#L233-L234
That value is always more accurate/correct than that serialization fallback, right?

Reply:
You're right, this can be disregarded!